samples/apps/copilot-chat-app/webapi/Skills/ChatSkill.cs

```diff
@@ -177,135 +178,92 @@ public async Task<SKContext> ChatAsync(string message, SKContext context)
+    /// Save a new message to the chat history.
+    /// This method will only modify the memory and the log of the context.
     /// </summary>
-    /// <param name="context">Contains the memory object.</param>
-    private async IAsyncEnumerable<MemoryQueryResult?> GetAllMemoriesAsync(SKContext context)
+    /// <param name="message">The message</param>
+    /// <param name="userId">The user ID</param>
+    /// <param name="userName"></param>
+    /// <param name="chatId">The chat ID</param>
+    private async Task SaveNewMessageAsync(string message, string userId, string userName, string chatId)
     {
-        var allCollections = await context.Memory.GetCollectionsAsync(context.CancellationToken);
-        var allChatMessageCollections = allCollections.Where(collection => collection != "ChatMessages");
-        IList<MemoryQueryResult> allChatMessageMemories = new List<MemoryQueryResult>();
-        try
-        {
-            foreach (var collection in allChatMessageCollections)
-            {
-                var results = await context.Memory.SearchAsync(
-                    collection,
-                    "abc", // dummy query since we don't care about relevance. An empty string will cause exception.
-                    limit: 1,
-                    minRelevanceScore: 0.0, // no relevance required since the collection only has one entry
-                    cancel: context.CancellationToken
-                ).ToListAsync();
-                allChatMessageMemories.Add(results.First());
-            }
-        }
-        catch (AIException ex)
-        {
-            context.Log.LogWarning("Exception while retrieving memories: {0}", ex);
-            context.Fail($"Exception while retrieving memories: {ex.Message}", ex);
-            yield break;
-        }
+        // Make sure the chat exists.
+        await this._chatRepository.FindById(chatId);
 
-        foreach (var memory in allChatMessageMemories.OrderBy(memory => memory.Metadata.Id))
-        {
-            yield return memory;
-        }
+        var messageId = Guid.NewGuid().ToString();
+        var chatMessage = new ChatMessage(messageId, userId, userName, chatId, message);
+        await this._chatMessageRepository.Create(chatMessage);
     }
 
     /// <summary>
-    /// Get the latest chat message from memory.
+    /// Save a new response to the chat history.
+    /// This method will only modify the memory and the log of the context.
     /// </summary>
-    /// <param name="context">Contains the memory object.</param>
-    private async Task<MemoryQueryResult?> GetLatestMemoryAsync(SKContext context)
+    /// <param name="response"></param>
+    /// <param name="chatId">The chat ID</param>
+    private async Task SaveNewResponseAsync(string response, string chatId)
     {
-        var allMemories = this.GetAllMemoriesAsync(context);
-
-        return await allMemories.FirstOrDefaultAsync();
+        await this.SaveNewMessageAsync(response, "bot", "bot", chatId);
```

Comment:
You could drop async from the method signature and just return this.SaveNewMessaegAsync

Reply:
This is no longer needed.